### ISSUE TITLE
Random Messages getting lost?

### ISSUE THREAD

**Issue by Mystro256** [emesene/emesene#813]
I seem to be not getting some of the messages sent to me, or at least not appearing in the conversation window. I know I getting them because the tray icon flashes and the messages are added to the chat history, but it's clearly not in the chat window :S

I can't pin point the issue but here's the debug from when it happened:

```
(11:21:29) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='a193bd98-656f-4a85-8e7e-ee81429f0689' network='1' account='s*******@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(11:21:29) papyon.transport:<<< MSG s*******@hotmail.com some stuff...
(11:21:32) papyon.protocol.switchboard_manager:Handler <SwitchboardConversation participants="s*******@hotmail.com"> attached to switchboard 840034491
(11:21:34) papyon.transport:<<< MSG s*******@hotmail.com some more stuff...
(11:22:37) papyon.protocol.switchboard:Disconnected (840034491)
```

Perhaps a similar issue was fixed recently? I'm using 2.11.7 at the moment. I'm also using papyon git, updated today.

**Comment by hit** [comment]
I'm also getting some random message loss with emmesene/papyone git. Tried to trace this on the other day, but with no luck

**Comment by c10ud** [comment]
i'd like a full debug but i fear is the same papyon issue where OUT doesn't really kill a switchboard -- it's difficult to reproduce

**Comment by Mystro256** [comment]
@bubuntux If so, it would make sense to just draw the message in the window before notifying no? Thus avoiding a loss of message if the notification doesn't work.

@c10ud All the debug I posted above is what seemed relevant. Although if I do manage to reproduce it I will post as much as possible.

I have found that I'm not getting my offline messages any more, as in the messages sent to me while not connected. I've been assured it's not a papyon glitch, and it's definitely not existent in empathy or pidgin. Perhaps it's related to this issue then? Since empathy uses papyon, it's safe to say it's something up with emesene.

**Comment by c10ud** [comment]
new papyon inside, not so much changed but there could be interesting ones

**Comment by hit** [comment]
Still the same, so back to Empathy for now.

**Comment by c10ud** [comment]
you may want to be aware that empathy uses papyon for the msn
connection, just a worser version than ours

2011/9/15 Rene Pärts
reply@reply.github.com:

**Comment by hit** [comment]
I know, but everything alright there (!)

**Comment by c10ud** [comment]
oh, interesting, might be interesting to debug if you can reproduce
systematically then

2011/9/15 Rene Pärts
reply@reply.github.com:

**Comment by Mystro256** [comment]
I've just done some testing with empathy, and I don't have that offline message glitch, interesting...

Can't seem to get a proper debug though...

**Comment by Sbte** [comment]
c10ud, if the message appears as a notification but not as a message in the window, I suspect it having something to do with cids or something?

I also think I have the offline message glitch, because people ask me why I didn't reply to them.

**Comment by c10ud** [comment]
oims: sometimes the server likes to answer with nice 500s, or even
your friends' client may be at fault

example: i know someone that from WLM sends OIMs to a person (using
adium) and it fscking fails from WLM side!

about the "normal" messages loss: we should check IF the message
received handler is called for this orphaned messages (without a
switchboard or something) and where is failing. please find a way to
reproduce this 100% and we'll fix it once and for all

2011/9/16 Sven reply@reply.github.com:

**Comment by Sbte** [comment]
Ok, go see if it's fixed...

**Comment by Mystro256** [comment]
Awesome, it fixed the offline message issue.... Although I'm still getting the lost message issue. Actually when I signed in (with my test account), I got a bunch of offline messages and though the convo that came up was all blank. It did appear in the history, thus this issue seems to still persist. Heres a full debug this time, all I did was log in and look at the history of conversation with my other account.

(10:59:00) papyon.transport:>>> MSG 60 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:00) papyon.transport:<<< ACK 56
(10:59:00) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f785d6d: blob 0, session 1650799947, data [1242 bytes](10:59:00) papyon.transport:>>> MSG 61 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:00) papyon.transport:<<< ACK 57
(10:59:00) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f78625b: blob 0, session 1650799947, data [1242 bytes](10:59:00) papyon.transport:>>> MSG 62 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:00) papyon.transport:<<< ACK 58
(10:59:00) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f786749: blob 0, session 1650799947, data [1242 bytes](10:59:00) papyon.transport:>>> MSG 63 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:00) papyon.transport:<<< ACK 59
(10:59:00) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f786c37: blob 0, session 1650799947, data [1242 bytes](10:59:00) papyon.transport:>>> MSG 64 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:00) papyon.transport:<<< ACK 60
(10:59:00) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f787125: blob 0, session 1650799947, data [1242 bytes](10:59:00) papyon.transport:>>> MSG 65 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:04) papyon.transport:<<< ACK 61
(10:59:04) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f787613: blob 0, session 1650799947, data [1242 bytes](10:59:04) papyon.transport:>>> MSG 66 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:04) papyon.transport:<<< ACK 62
(10:59:04) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f787b01: blob 0, session 1650799947, data [1242 bytes](10:59:04) papyon.transport:>>> MSG 67 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:04) papyon.transport:<<< ACK 63
(10:59:04) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f787fef: blob 0, session 1650799947, data [1242 bytes](10:59:05) papyon.transport:>>> MSG 68 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:05) papyon.transport:<<< ACK 64
(10:59:05) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f7884dd: blob 0, session 1650799947, data [1242 bytes](10:59:05) papyon.transport:>>> MSG 69 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:05) papyon.transport:<<< ACK 65
(10:59:05) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f7889cb: blob 0, session 1650799947, data [1242 bytes](10:59:05) papyon.transport:>>> MSG 70 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (1270 bytes)](10:59:05) papyon.transport:<<< ACK 66
(10:59:05) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f788eb9: blob 0, session 1650799947, data [896 bytes](10:59:05) papyon.transport:>>> MSG 71 D 249
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (912 bytes)](10:59:05) papyon.transport:<<< ACK 67
(10:59:05) papyon.transport:<<< ACK 68
(10:59:05) papyon.transport:<<< ACK 69
(10:59:05) papyon.transport:<<< ACK 70
(10:59:05) papyon.transport:<<< ACK 71
(10:59:05) papyon.msnp2p.session:Session data transfer completed
(10:59:05) papyon.transport:<<< MSG alexjnewt<at>hotmail.com alexjnewt<at>hotmail.com 249
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    P2P-Dest: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    [P2P message (512 bytes)](10:59:05) papyon.msnp2p.transport.switchboard:<<< TLPv2 chunk 0x4b56ed56: blob 0, RAK, SLP Message
    BYE MSNMSGR:alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d} MSNSLP/1.0\r\n
    To: msnmsgr:alexjnewt<atlive.com;{00130ccc-c64f-492e-829c-cfa13cef104d}>\r\n
    From: msnmsgr:alexjnewt<athotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}>\r\n
    Via: MSNSLP/1.0/TLP ;branch={068bc044-24d0-4551-bc18-dcb06d8a6828}\r\n
    CSeq: 0\r\n
    Call-ID: {ba6a1c7a-00aa-4cb0-8c97-f89015b7708a}\r\n
    Max-Forwards: 0\r\n
    Content-Type: application/x-msnmsgr-sessionclosebody\r\n
    Content-Length: 44\r\n
    \r\n
    SessionID: 1650799947\r\n
    SChannelState: 0\r\n
    \r\n
    \x00
(10:59:05) papyon.msnp2p.transport.switchboard:>>> TLPv2 chunk 0x6f789241: blob 0, ACK 0x4b56ef4e,
(10:59:05) papyon.transport:>>> MSG 72 D 248
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'alexjnewt<at>live.com;{00130ccc-c64f-492e-829c-cfa13cef104d}'\r\n
    P2P-Dest: 'alexjnewt<at>hotmail.com;{602b66c6-e3f1-4685-b45a-16679db00e3b}'\r\n
    [P2P message (16 bytes)](10:59:05) papyon.msnp2p.session:Session 1650799947 disposed
(10:59:05) papyon.msnp2p.transport:Cleaning up session 1650799947
(10:59:05) papyon.transport:<<< ACK 72
(10:59:25) papyon.transport:<<< CHL 0 13506952491421915918
(10:59:25) papyon.transport:>>> QRY 13 PROD0114ES4Z%Q5W 32
'd85587defc1ef078af2eddb0cd1e9fb3'
(10:59:25) papyon.transport:<<< QRY 13
(10:59:40) papyon.transport:>>> PNG
(10:59:40) papyon.transport:<<< QNG 46

**Comment by Mystro256** [comment]
the @live.com (testing account) being the account I tested on the receiving end and the @hotmail.com (main account) being the sending

**Comment by Sbte** [comment]
It only happens to oims?

**Comment by Sbte** [comment]
Oh, and they work fine for me...

**Comment by c10ud** [comment]
> I got a bunch of offline messages and though the convo that came up was all blank.

Different issue than OPs

> It did appear in the history, thus this issue seems to still persist. Heres a full debug this time, all I did was log in and look at the history of conversation with my other account.

The debug only shows a data transfer

**Comment by Mystro256** [comment]
@Sbte no, but I figured it was related because it was the same symptoms :/

The issue seems to happen at random for me, but it just by chance happened when I got those offline messages. When it happened to me before, it was mid conversation

@c10ud well the issue, if its not clear (I am the OP ;) lol), was that the messages weren't coming up in the conversation window but were coming up in the history only, or at least that's what I observed. When I got those offline messages, they came up in the history but not the conversation window. I did not do any further testing, but I the symptoms seemed the same, thus I posted the debug right away in hopes to fix this.

**Comment by Sbte** [comment]
Maybe it's webkit that's hanging? I can think of nothing else...

**Comment by c10ud** [comment]
needs to be rechecked as soon as the 2.11.11 binary is up

**Comment by c10ud** [comment]
woah wrong bug -- this issue is emesene specific, the message never reaches the ConversationManager so we're either have some zombie ConversationManager or something like that!

**Comment by c10ud** [comment]
OR

conv_first_action is not issued because conv is in self.rpapyconv

**Comment by c10ud** [comment]
..but it really isn't

**Comment by Sbte** [comment]
Just fixed another bug related to this. Not sure if it really fixed this issue this time.

**Comment by c10ud** [comment]
If you could try with current git master there have been modifications regarding this issue

**Comment by daniid** [comment]
This is still happenning in the version I downloaded 3 days ago (emesene-emesene-v2.11.11-42-gb057d01.zip)

I will try posting more details and a log as soon as I can.

**Comment by daniid** [comment]
I'll post a log in case it turns out to be useful:
- Using emesene-emesene-v2.11.11-42-gb057d01
- Kubuntu 10.10, kernel 2.6.35-30-generic
- KDE 4.5.5, Qt 4.7.0

I was talking to my friend buddy_1, then stopped talking for a while, then talked again:
- I sent a message at (18:23:05)
- I got a response at (18:23:15) but it didn't turn up in the conversation window. It was properly logged, though.
- At (18:32:40) I realised the conversation wasn't working and logged out.

```
(18:22:32) papyon.transport:>>> PNG
(18:22:33) papyon.transport:<<< QNG 50
(18:22:51) papyon.transport:>>> PNG
(18:22:52) papyon.transport:<<< QNG 41
(18:23:03) papyon.protocol.switchboard_manager:Requesting switchboard for participant(s) buddy_1@hotmail.com
(18:23:03) papyon.protocol.switchboard_manager:Requesting new switchboard
(18:23:03) papyon.transport:>>> XFR 18 SB
(18:23:03) papyon.transport:<<< XFR 18 SB 64.4.61.155:1863 CKI 785523378.2349428.7618512 U messenger.msn.com 1
(18:23:03) papyon.protocol.switchboard:New switchboard session 785523378.2349428.7618512
(18:23:03) papyon.transport:<-> Connecting to 64.4.61.155:1863
(18:23:03) papyon.transport:>>> USR 0 my_account@hotmail.com;{75fbb861-b753-4242-a95b-3c9ed045a63b} 785523378.2349428.7618512
(18:23:04) papyon.transport:<<< USR 0 OK my_account@hotmail.com;{75fbb861-b753-4242-a95b-3c9ed045a63b} Dani
(18:23:04) papyon.protocol.switchboard_manager:Handler <SwitchboardConversation participants="buddy_1@hotmail.com"> attached to switchboard 785523378.2349428.7618512
(18:23:04) papyon.transport:>>> CAL 1 my_account@hotmail.com
(18:23:04) papyon.transport:>>> CAL 2 buddy_1@hotmail.com
(18:23:04) papyon.transport:<<< CAL 1 RINGING 785523378
(18:23:04) papyon.transport:<<< RNG 785523378 64.4.61.155:1863 CKI 18833202.5036184.1997087827 my_account@hotmail.com Dani U messenger.msn.com 1
(18:23:04) papyon.protocol.switchboard:New switchboard session 785523378
(18:23:04) papyon.transport:<-> Connecting to 64.4.61.155:1863
(18:23:04) papyon.transport:>>> ANS 0 my_account@hotmail.com;{209f478c-6fd8-43e0-a636-44f1cb727831} 18833202.5036184.1997087827 785523378
(18:23:05) papyon.transport:<<< IRO 0 1 1 my_account@hotmail.com;{75fbb861-b753-4242-a95b-3c9ed045a63b} Dani 2751463680:3624129568
(18:23:05) papyon.transport:<<< ANS 0 OK
(18:23:05) papyon.transport:<<< JOI my_account@hotmail.com Dani 2751463680:0
(18:23:05) papyon.transport:<<< JOI my_account@hotmail.com;{209f478c-6fd8-43e0-a636-44f1cb727831} Dani 2751463680:0
(18:23:05) papyon.transport:<<< JOI my_account@hotmail.com Dani 2751463680:0
(18:23:05) papyon.transport:<<< NLN NLN 1:buddy_3@hotmail.com EL%20TIO 3055849760:2609258384 %3cmsnobj%20Creator%3d%22buddy_3%40hotmail.com%22%20Size%3d%2229763%22%20Type%3d%223%22%20SHA1D%3d%22olza8uOGr%2f%2bp%2f%2fAgVnbQ1hJDucw%3d%22%20Url%3d%22%22%20Url1%3d%22http%3a%2f%2flivefiles18.vo.msecnd.net%2fy1mY4-0itMEN1rgwiQbvlY0T6znVnnuTX43IypKP05qCJT2u0w3DrTWG5GEg96F1zAyNa1efvC2UZABH185Yg9gS5Gj1M5opT1g%22%2f%3e
(18:23:05) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='147a5d73-0000-0000-0000-000000000000' network='1' account='buddy_3@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:23:05) papyon.transport:<<< CAL 2 RINGING 785523378
(18:23:05) papyon.transport:<<< NLN NLN 1:buddy_3@hotmail.com EL%20TIO 3055849760:2609258384 %3cmsnobj%20Creator%3d%22buddy_3%40hotmail.com%22%20Size%3d%2229763%22%20Type%3d%223%22%20SHA1D%3d%22olza8uOGr%2f%2bp%2f%2fAgVnbQ1hJDucw%3d%22%20Url%3d%22%22%20Url1%3d%22http%3a%2f%2flivefiles18.vo.msecnd.net%2fy1mY4-0itMEN1rgwiQbvlY0T6znVnnuTX43IypKP05qCJT2u0w3DrTWG5GEg96F1zAyNa1efvC2UZABH185Yg9gS5Gj1M5opT1g%22%2f%3e
(18:23:05) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='147a5d73-0000-0000-0000-000000000000' network='1' account='buddy_3@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:23:05) papyon.transport:<<< UBX 1:buddy_3@hotmail.com 793
    [payload]
(18:23:05) papyon.profile:Set contact buddy_3@hotmail.com flags to 1
(18:23:05) papyon.transport:<<< UBX 1:buddy_3@hotmail.com 793
    [payload]
(18:23:05) papyon.profile:Set contact buddy_3@hotmail.com flags to 1
(18:23:05) papyon.transport:<<< JOI buddy_1@hotmail.com;{fa2229dd-5bdb-4a80-b9dd-77cb37282475} Squall 2788999484:2550273072
(18:23:05) papyon.transport:<<< JOI buddy_1@hotmail.com Squall 2788999484:2550273072
(18:23:05) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='22b55cef-4d3d-424c-9398-6201bb2c83f9' network='1' account='buddy_1@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:23:05) papyon.transport:<<< JOI buddy_1@hotmail.com;{fa2229dd-5bdb-4a80-b9dd-77cb37282475} Squall 2788999484:2550273072
(18:23:05) papyon.transport:<<< JOI buddy_1@hotmail.com Squall 2788999484:2550273072
(18:23:05) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='22b55cef-4d3d-424c-9398-6201bb2c83f9' network='1' account='buddy_1@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:23:05) papyon.transport:>>> MSG 3 N 187
    X-MMS-IM-Format: 'FN=Sans; EF=; CO=702b00; CS=1; PF=00'\r\n
    Content-Type: 'text/plain; charset=utf-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    env\xc3\xado un sms ...?
(18:23:05) papyon.transport:<<< MSG my_account@hotmail.com Dani 187
    X-MMS-IM-Format: 'FN=Sans; EF=; CO=702b00; CS=1; PF=00'\r\n
    Content-Type: 'text/plain; charset=utf-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    env\xc3\xado un sms ...?
(18:23:05) papyon.protocol.switchboard_manager:Handler <SwitchboardConversation participants="buddy_1@hotmail.com"> attached to switchboard 785523378
(18:23:14) papyon.transport:<<< MSG buddy_1@hotmail.com Squall 116
    TypingUser: 'buddy_1@hotmail.com'\r\n
    Content-Type: 'text/x-msmsgscontrol'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    \r\n
(18:23:14) papyon.transport:<<< MSG buddy_1@hotmail.com Squall 116
    TypingUser: 'buddy_1@hotmail.com'\r\n
    Content-Type: 'text/x-msmsgscontrol'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    \r\n
(18:23:15) papyon.transport:<<< MSG buddy_1@hotmail.com Squall 144
    X-MMS-IM-Format: 'FN=Arial; EF=B; CO=a00000; CS=0; PF=22'\r\n
    Content-Type: 'text/plain; charset=UTF-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    ok
(18:23:15) papyon.transport:<<< MSG buddy_1@hotmail.com Squall 144
    X-MMS-IM-Format: 'FN=Arial; EF=B; CO=a00000; CS=0; PF=22'\r\n
    Content-Type: 'text/plain; charset=UTF-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    ok
(18:23:22) papyon.transport:>>> PNG
(18:23:23) papyon.transport:<<< QNG 45
(18:23:32) papyon.transport:>>> PNG
(18:23:33) papyon.transport:<<< QNG 50
(18:24:07) papyon.transport:>>> PNG
(18:24:08) papyon.transport:<<< QNG 49
(18:24:22) papyon.transport:>>> PNG
(18:24:23) papyon.transport:<<< QNG 42
(18:24:56) papyon.transport:>>> PNG
(18:24:57) papyon.transport:<<< QNG 48
(18:25:05) papyon.transport:>>> PNG
(18:25:06) papyon.transport:<<< QNG 41
(18:25:44) papyon.transport:>>> PNG
(18:25:45) papyon.transport:<<< QNG 50
(18:25:46) papyon.transport:>>> PNG
(18:25:47) papyon.transport:<<< QNG 44
(18:26:31) papyon.transport:>>> PNG
(18:26:32) papyon.transport:<<< QNG 47
(18:26:34) papyon.transport:>>> PNG
(18:26:35) papyon.transport:<<< QNG 44
(18:27:18) papyon.transport:>>> PNG
(18:27:18) papyon.transport:>>> PNG
(18:27:19) papyon.transport:<<< QNG 46
(18:27:19) papyon.transport:<<< QNG 39
(18:27:38) papyon.transport:<<< UBX 1:buddy_2@gmail.com 377
    [payload]
(18:27:38) papyon.profile:Set contact buddy_2@gmail.com flags to 1
(18:27:38) papyon.transport:<<< UBX 1:buddy_2@gmail.com 377
    [payload]
(18:27:38) papyon.profile:Set contact buddy_2@gmail.com flags to 1
(18:27:38) papyon.transport:<<< NLN AWY 1:buddy_2@gmail.com José 2416181284:2147619840 0
(18:27:38) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='9eadc523-9f52-4bfa-885f-bf1e6f7a798c' network='1' account='buddy_2@gmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:27:38) papyon.transport:<<< NLN AWY 1:buddy_2@gmail.com José 2416181284:2147619840 0
(18:27:38) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='9eadc523-9f52-4bfa-885f-bf1e6f7a798c' network='1' account='buddy_2@gmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(18:27:57) papyon.transport:>>> PNG
(18:27:58) papyon.transport:<<< QNG 51
(18:28:04) papyon.transport:>>> PNG
(18:28:05) papyon.transport:<<< QNG 42
(18:28:15) papyon.protocol.switchboard:Switchboard timed out. Going to leave it.
(18:28:15) papyon.protocol.switchboard:Leaving switchboard 785523378
(18:28:15) papyon.transport:>>> OUT
(18:28:15) papyon.protocol.switchboard:Switchboard timed out. Going to leave it.
(18:28:15) papyon.protocol.switchboard:Leaving switchboard 785523378.2349428.7618512
(18:28:15) papyon.transport:>>> OUT
(18:28:16) papyon.protocol.switchboard:Disconnected (785523378.2349428.7618512)
(18:28:16) papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="buddy_1@hotmail.com">
(18:28:16) papyon.transport:<<< BYE my_account@hotmail.com;{75fbb861-b753-4242-a95b-3c9ed045a63b}
(18:28:16) papyon.protocol.switchboard:Disconnected (785523378)
(18:28:16) papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="buddy_1@hotmail.com">
(18:28:46) papyon.transport:>>> PNG
(18:28:47) papyon.transport:<<< QNG 48
(18:28:48) papyon.transport:>>> PNG
(18:28:49) papyon.transport:<<< QNG 44
(18:29:32) papyon.transport:>>> PNG
(18:29:33) papyon.transport:<<< QNG 40
(18:29:34) papyon.transport:>>> PNG
(18:29:35) papyon.transport:<<< QNG 49
(18:30:12) papyon.transport:>>> PNG
(18:30:13) papyon.transport:<<< QNG 41
(18:30:23) papyon.transport:>>> PNG
(18:30:24) papyon.transport:<<< QNG 50
(18:30:53) papyon.transport:>>> PNG
(18:30:54) papyon.transport:<<< QNG 44
(18:31:13) papyon.transport:>>> PNG
(18:31:14) papyon.transport:<<< QNG 46
(18:31:37) papyon.transport:>>> PNG
(18:31:38) papyon.transport:<<< QNG 46
(18:31:59) papyon.transport:>>> PNG
(18:32:00) papyon.transport:<<< QNG 40
(18:32:23) papyon.transport:>>> PNG
(18:32:24) papyon.transport:<<< QNG 40
(18:32:39) papyon.transport:>>> PNG
(18:32:40) papyon.transport:<<< QNG 43
```

This has happened in single person and group conversations.

**Comment by c10ud** [comment]
do you know which client your friend is using?

**Comment by daniid** [comment]
Yes, he's using the official WLM client, apparently the about dialog says it's "14.0.8117.416 version 2009".

**Comment by c10ud** [comment]
@Sbte 's modifications about being more strict with switchboard and me increasing the switchboard timeout to 5 minutes could have affected this issue, needs to be checked but the low noise on the bug makes me wonder if we finally have it fixed in git master (?)

**Comment by Sbte** [comment]
I don't know. If it's really a gui bug, then it could be, but I'm not sure.

**Comment by Stiveknx** [comment]
After install the LastSaid plugin, some messages started to get lost here ...
I'm not sure if really is the plugin or another plugin, and I'm waiting for the bug to happens to create an issue.

I see the messages in debug, but they just didn't show in conversation window.

**Comment by daniid** [comment]
I stopped reporting since I went back to the official web-based messenger, but I'll gladly test the latest commits :)

**Comment by Stiveknx** [comment]
With LastSaid plugin enabled:
http://i.minus.com/irUkPjfxnFv2X.png

The message came, I saw in notify-osd but I didn't saw in conversation window.
There's no error in debug,
In info:

(12:29:07) papyon.protocol.switchboard_manager:Handler <SwitchboardConversation participants="miojex360@hotmail.com"> attached to switchboard 36416946
(12:29:42) papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="miojex360@hotmail.com">
(12:29:42) papyon.protocol.switchboard:Leaving switchboard 36416946
(12:29:42) papyon.protocol.switchboard:Disconnected (36416946)
(12:29:44) papyon.protocol.switchboard:New switchboard session 1813155542
(12:29:44) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='76c8502a-0000-0000-0000-000000000000' network='1' account='miojex360@hotmail.com' memberships='FORWARD | ALLOW'>

**Comment by c10ud** [comment]
http://blog.emesene.org/

Known issues:

```
the last said plugin breaks the conversation, just don't use it for now
```

**Comment by Sbte** [comment]
Might be fixed with current master and the latest version of the last said plugin, because we made some updates to it.

**Comment by Stiveknx** [comment]
Didn't happen here anymore after c8fefb5cd04c3532f204da3c61b65cbb08e5a89c which also updated last said plugin.
But there's another bug, when you logout and login in again, if you have two or more conversations, in window 1 you see the conversation from window 2 and vice-versa. I'll create an issue latter maybe for that, if this happens again

**Comment by Sbte** [comment]
4a8c5c3e31479b19de141b1a847deacd89042d0d

This should at least help with the lost messages. Not sure it the issue is really fixed this time, but time will tell...

**Comment by daniid** [comment]
The problem (the one I'm experiencing anyway) keeps happening. 

My specs:
- Kubuntu 10.10, KDE 4.5.5, kernel 2.6.35-31-generic
- Source zipball: emesene-emesene-e91a1a8 
- Enabled plug-ins: contact status history combo. (I read about the "last said" plugin bug but I never even downloaded it)

This is the log of the conversation according to the history (Right click on contact's name > See information > Conversation history). This happened after chatting for a few minutes:
```dom 12 feb 2012 20:30:49 WET Myself: blabla1
dom 12 feb 2012 20:36:05 WET Myself: btw blabla?
dom 12 feb 2012 20:36:13 WET My_friend: yes
dom 12 feb 2012 20:50:20 WET Myself: blabla2
dom 12 feb 2012 20:50:24 WET Myself: blabla3
dom 12 feb 2012 20:50:55 WET My_friend: blabla4

```

I didn't get (i.e., they didn't turn up in the conversation window despite appearing in the log) any of my friend's messages after, and including, the one at 20:36:13. Notice that none of us said anything for 5 minutes at first, probably causing some kind of inactivity that wasn't handled properly.

I saved the debug log so I'll upload it if necessary. 
```

**Comment by Sbte** [comment]
@daniid Are you using Text Output? i.e., no fancy colors?

**Comment by daniid** [comment]
Do you mean Text Output vs. Adium Output? I don't even seem to have a choice. There's no "conversation output" category in the plug-in menu, so I'm using the default output with no fancy colors or graphics, I guess that's Text Output by default.

I remember this same issue happening in previous versions (when I did have a choice in the plug-in menu) with both Text and Adium output.

**Comment by c10ud** [comment]
me changing the conversation timeout to 5mins and sbte fixes _should_ have finally fixed this in 2.12.x
